tests/sentry/incidents/endpoints/test_project_alert_rule_details.py

```diff
@@ -226,8 +278,115 @@ def test_kicks_off_slack_async_job(
+        channelID = 123
+        associated_channel_name = "some-other-channel"
+        test_params = self._slack_setup_helper(channelName, channelID)
+
+        resp = self._slack_response_mocking(
+            url=f"https://slack.com/api/conversations.info?channel={channelID}",
+            body={"ok": "true", "channel": {"name": associated_channel_name}},
+            params=test_params,
+        )
+        assert resp.status_code == 400
+        assert resp.data == {
+            "nonFieldErrors": [
+                ErrorDetail(
+                    string="Received channel name some-other-channel does not match inputted channel name my-channel.",
+                    code="invalid",
+                )
+            ]
+        }
+        # XXX: What message does the UI show?
+
+    @responses.activate
+    def test_create_slack_alert_with_non_existent_channel_id(self):
+        """
+        The user specifies a bad Slack channel ID.
+        """
+        channelName = "my-channel"
+        # Specifying an inputChannelID will cause the validate_channel_id logic to be triggered
+        channelID = 123
+        test_params = self._slack_setup_helper(channelName, channelID)
+
+        from sentry.integrations.slack.client import SlackClient
+
+        # An incorrect channelID will raise an ApiError in the Slack client
+        with patch.object(SlackClient, "get", side_effect=ApiError(text="channel_not_found")):
+            with self.feature("organizations:incidents"):
+                # Within validate_channel_id this error will be raised
+                # ValidationError("Channel not found. Invalid ID provided.")
+                resp = self.get_response(
+                    self.organization.slug, self.project.slug, self.alert_rule.id, **test_params
+                )
+        assert resp.status_code == 400
+        assert resp.data == {
+            "nonFieldErrors": [
+                ErrorDetail(string="Channel not found. Invalid ID provided.", code="invalid")
+            ]
+        }
+
+    @patch("sentry.tasks.integrations.slack.find_channel_id_for_alert_rule.apply_async")
+    @patch("sentry.integrations.slack.utils.rule_status.uuid4")
+    @responses.activate
+    def test_create_slack_alert_with_empty_channel_id(
+        self, mock_uuid4, mock_find_channel_id_for_alert_rule
+    ):
+        """
+        The user selects the channel ID field and the UI will send the empty string to the
+        endpoint, thus, a channel name search will be performed
+        """
+        mock_uuid4.return_value = self.get_mock_uuid()
+        channelName = "my-channel"
+        # Because channel ID is an empty string it will be converted to an async request for the channel name
+        test_params = self._slack_setup_helper(channelName, channelID="")
+
+        with self.feature("organizations:incidents"):
+            # Within validate_channel_id this error will be raised
```

Comment:
Could we make it so an empty string doesn't trigger channel ID validation?

Reply:
My apologies. That comment was from copy/pasting. I've verified that the validation code is not executed with this test.